### PR TITLE
[sw/silicon_creator] Update manifest fields

### PR DESCRIPTION
**Pull request by alphan**
This change:
- Removes `extension_*`, `reserved*`, `usage_constraints`, `lockdown_info`, `image_version`,
- Adds a struct for key manager binding value (`keymgr_binding_value_t`), and
- Adds `image_major_version`, `image_minor_version`, `binding_value`, and `max_key_version` fields.

@moidx I saw #5956 but the last time we met the final format that we came up with did not include `usage_constraints`. Please let me know if you would like me to keep it.

### CODE REVIEW

File: sw/device/silicon_creator/lib/drivers/keymgr.c [hunk 1]
```diff
@@ -86,7 +86,7 @@ rom_error_t keymgr_state_advance_to_creator(const uint32_t binding_value[8],
   for (size_t i = 0; i < 8; ++i) {
```
Comment (cfrantz) [inline sw/device/silicon_creator/lib/drivers/keymgr.c:86]: Should we use `ARRAYSIZE(binding_value->data)` ?

File: sw/device/silicon_creator/lib/keymgr_binding_value.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Binding value used by key manager to derive secret values.
+ *
+ * A change in this value changes the secret value of key manager, and
+ * consequently, the versioned keys and identity seeds generated at subsequent
+ * boot stages.
+ */
+typedef struct keymgr_binding_value {
+  uint32_t data[8];
```
Comment (cfrantz) [inline sw/device/silicon_creator/lib/keymgr_binding_value.h:22]: Where does the size of 8 words come from?  (Looks like this is an implementation detail of the hardware: it gives you an 8-word binding value.  Suggest adding a comment that says so.)

**Review by cfrantz — APPROVED** [review]
Is there any opentitan.org documentation that should be updated?

File: sw/device/silicon_creator/lib/drivers/keymgr.c (hunk 1 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/drivers/keymgr.c:86]: Yes, done.

File: sw/device/silicon_creator/lib/keymgr_binding_value.h (hunk 2 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/keymgr_binding_value.h:22]: Done.

**Comment by alphan** [comment]
> Is there any opentitan.org documentation that should be updated?

`sw/device/silicon_creator/rom_exts/docs/manifest.md` should be updated. I'll update that in a separate PR.

**Review by moidx — APPROVED** [review]

**Comment by moidx** [comment]
Hi @alphan, the usage constraint requirements described in #5956 will require defining new manifest fields. I am ok with removing the current `usage_constraints` field as part of this PR.

**Comment by alphan** [comment]
> Hi @alphan, the usage constraint requirements described in #5956 will require defining new manifest fields. I am ok with removing the current `usage_constraints` field as part of this PR.

Sounds good, thanks @moidx!

**Pull request merged**